lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+  pagenation.classList.add("slider__pagination");
+
+  let fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.classList.add("current");
+  current.textContent = 1;
+  fragment.appendChild(current);
+
+  const separation = document.createElement("span");
+  separation.textContent = "/";
+  separation.classList.add("separation");
+  fragment.appendChild(separation);
+
+  const total = document.createElement("span");
+  total.classList.add("total");
+  total.textContent = item.length;
+  fragment.appendChild(total);
+
+  const getTextArea = document.querySelector(".mainvisual__textarea");
+  getTextArea.appendChild(pagenation).appendChild(fragment);
+}
+
+function pagenationCountDown() {
+  const pagenationCurrent = document.querySelector(".current");
+  const pagenationTotal = Number(document.querySelector(".total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  if (currentNum <= pagenationTotal && currentNum !== 1) {
+    pagenationCurrent.textContent = currentNum -= 1;
+  }
+}
+
+function pagenationCountUp() {
+  const pagenationCurrent = document.querySelector(".current");
+  const pagenationTotal = Number(document.querySelector(".total").textContent);
+  let currentNum = Number(pagenationCurrent.textContent);
+
+  if (currentNum <= pagenationTotal && currentNum !== 5) {
+    pagenationCurrent.textContent = currentNum += 1;
+  }
+}
+
+function ImageNextSlide(e) {
+  const slideImage = document.querySelectorAll(".slider__item");
+  const activeSlide = document.querySelector(".is-active");
+  const nextSlide = activeSlide.nextElementSibling;
+  const nextSlideIndex = Number(nextSlide.getAttribute("data-slide-index"));
+  if (nextSlideIndex === slideImage.length) {
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.previousElementSibling.disabled = false;
+    nextSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
+
+function ImagePrevSlide(e) {
+  const activeSlide = document.querySelector(".is-active");
+  const prevSlide = activeSlide.previousElementSibling;
+  const prevSlideIndex = Number(prevSlide.getAttribute("data-slide-index"));
+
+  if (prevSlideIndex === 1) {
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+    e.disabled = true;
+  } else {
+    e.nextElementSibling.disabled = false;
+    prevSlide.classList.add("is-active");
+    activeSlide.classList.remove("is-active");
+  }
+}
+
+function callImageData() {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(fetchImageData());
+    }, 3000);
+  });
+}
+
+async function fetchImageData() {
+  const SLIDE_DATE_URL =
+    "https://api.json-generator.com/templates/9tm12BO1y5Xx/data?access_token=b0154huvd1stffra1six9olbgg34r4zofcqgwzfl&delay=1000";
+  const response = await fetch(SLIDE_DATE_URL);
```

Comment:
IMO
Moriken-sensei told me that I should write it in the function caller
```suggestion
function callImageData() {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve(fetchImageData(URL));　//here
    }, 3000);
  });
}
```

Reply:
Entering the URL at the caller makes sense for the function!
informative.Thank you!

[9022835](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/9022835c65b538690f8dfe7939a485f797b649f1)